sympy/series/series.py

```diff
@@ -10,3 +13,133 @@ def series(expr, x=None, x0=0, n=6, dir="+"):
+    This is also called reversion of series.
+
+    Parameters
+    ==========
+
+    eq : expression, the expansion of whose inverse is to be found.
+
+    no_of_terms : no of terms required in the expansion; default value is 3 terms.
+
+    Examples
+    ========
+
+    >>> from sympy.series.series import lagrange_inversion_theorem
+    >>> from sympy import exp
+    >>> from sympy.abc import x
+
+    Let the equation whose inverse's expansion is to be found be y = f(x)
+    then the first argument to be supplied must be f(x).
+    Like here, the equation whose inverse's expansion to be found is y = (x**2)*exp(x)
+
+    >>> eq = (x**2)*exp(x)
+
+    If no second argument is provided, the default number of terms i.e. 3, is present
+    in the expansion of the inverse of the provided equation or expression (as the first argument).
+
+    >>> lagrange_inversion_theorem(eq)
+    2*(x**2*exp(x) - E)**3*exp(-3)/27 - 7*(x**2*exp(x) - E)**2*exp(-2)/54 + (x**2*exp(x) - E)*exp(-1)/3
+
+    Here the number of terms required in the expansion is provided.
+
+    >>> lagrange_inversion_theorem(eq, 1)
+    (x**2*exp(x) - E)*exp(-1)/3
+    >>> lagrange_inversion_theorem(eq, 2)
+    -7*(x**2*exp(x) - E)**2*exp(-2)/54 + (x**2*exp(x) - E)*exp(-1)/3
+
+    >>> eq = exp(x)
+    >>> lagrange_inversion_theorem(eq, 2)
+    -(exp(x) - 1)**2/2 + exp(x) - 1
+    >>> lagrange_inversion_theorem(eq)
+    (exp(x) - 1)**3/3 - (exp(x) - 1)**2/2 + exp(x) - 1
+
+    Note
+    ====
+
+    There is a straightforward derivation using complex analysis and contour integration;
+    the complex formal power series version is a consequence of knowing the formula for polynomials,
+    so the theory of analytic functions may be applied. Actually, the machinery from analytic
+    function theory enters only in a formal way in this proof, in that what is really needed is
+    some property of the formal residue, and a more direct formal proof is available.
+
+    References
+    ==========
+    .. [1] https://en.wikipedia.org/wiki/Lagrange_inversion_theorem
+    .. [2] http://mathworld.wolfram.com/LagrangeInversionTheorem.html
+    .. [3] http://www.cfm.brown.edu/people/dobrush/am33/Mathematica/lit.html
+    """
+    from sympy.abc import w, x
+
+    try:
+        eq.free_symbols
+    except:
+        raise AssertionError("The function provided is not analytic i.e. f' = 0. Lagrange's inversion theorem "
+                             "requires that the function be analytic")
+    if len(eq.free_symbols) == 2:
+        if not (x in eq.free_symbols and w in eq.free_symbols):
+            raise AssertionError("The function provided must be in terms of 'x' or 'w'")
+    if len(eq.free_symbols) == 1:
+        if not (x in eq.free_symbols or w in eq.free_symbols):
+            raise AssertionError("The function provided must be in terms of 'x' or 'w'")
+
+    n = no_of_terms
+    eq = eq.subs(x, w)
+    z = eq
+
+    z_diff = diff(eq, w)
+    t = 0
```

Comment:
Instead of trying integer values, which might not succeed, this should also be provided as a parameter to the function.

Reply:
The user might not always be sure of taking the trouble of finding `a` and checking if it's a valid position to be applied for finding the expansion. This is why I took that route. But your suggestion made more sense when I dived into the actual documentation given on Wolfram as the expansion will depend on the point where it is being calculated. So thanks for the input. Have implemented now.